api/src/main/java/pl/touk/nussknacker/engine/api/AdditionalVariable.java

```diff
@@ -12,4 +12,6 @@
     String name();
 
     Class clazz();
+
+    boolean initializedByRuntime() default false;
```

Comment:
I see problems with this name of parameter:
1. Consider situation: someone provide additional variable "#MATH` and value of parameter `#MATH.pow(2, 2)` and some dependent parameters based on this value. How it is initialized by runtime? It will be initialized on designer side during validation and in runtime during compilation.
2. AdditionalVariableProvided*InRuntime* is opposite to `initializedByRuntime = true` and name is quite similar.
Maybe we should name this `initializedDuringCompilation`? Any other proposal?

Reply:
I think we don't need this flag at all :) If the parameter is lazy, then it's always false, if it's eager - it's always true...

Reply:
good point, let's change it that way - at the end the whole eager parameter is "initialized during compilation" ;)